experimental/segmenter/src/provider.rs

```diff
@@ -243,3 +243,24 @@ impl ResourceProvider<SentenceBreakDataV1Marker> for RuleBreakDataProvider {
         })
     }
 }
+
+/// char16trie data for dictionary break
+#[icu_provider::data_struct(UCharDictionaryBreakDataV1Marker = "segmenter/char16trie@1")]
+pub struct UCharDictionaryBreakDataV1<'data> {
+    /// Dictionary data of char16trie.
+    #[cfg_attr(feature = "provider_serde", serde(borrow))]
+    pub trie_data: ZeroVec<'data, u16>,
+}
+
+// Test data of thai dictionary
+const THAI_DICTIONARY: &[u8; 224978] = include_bytes!("../tests/testdata/thai.dict");
+
+impl<'data> Default for UCharDictionaryBreakDataV1<'data> {
+    fn default() -> Self {
+        let trie_data =
+            unsafe { core::mem::transmute::<&[u8; 224978], &[u16; 112489]>(THAI_DICTIONARY) };
+        Self {
+            trie_data: ZeroVec::from_slice(trie_data),
+        }
```

Comment:
And then down here, you can get rid of the `unsafe`

```suggestion
        Self {
            trie_data: ZeroVec::Borrowed(THAI_DICTIONARY),
        }
```

Reply:
Thanks